concurrent/actor/eventloop/EventLoop.java

```diff
@@ -53,6 +53,10 @@ public void submit(Runnable job) {
         return new ScheduledJob(millis, job);
     }
 
+    public void await() throws InterruptedException {
```

Comment:
`EventLoop` already has a `stop` method. Is that what you want? If you only `join` on the event loop thread, it will never quit the loop.

Reply:
Await is useful to block the main thread: 

```
public void main(String[] args) {
  EventLoopGroup elg = new EventLoopGroup(...);
  Actor<ActorRoot> rootActor = ...
  rootActor.tell(root -> ...)
  rootActor.tell(root -> ...)
  ...
  elg.await();
}
```

Reply:
The reason we never need any await method on our event loop in cluster is because we "piggyback" on the fact that we're using gRPC's `await` method which allows us to block the main thread.

Regardless, an asynchronous library should always have a way to await on the main thread.

Reply:
@lolski The thing is if we simply `await` on main thread, and the event loops are blocked waiting for new events, then both threads will be blocked forever? Is this actually what you want?

Reply:
Are you talking about this scenario?

```
public void main(String[] args) {
  EventLoopGroup elg = new EventLoopGroup(...);
  Actor<ActorRoot> rootActor = ...
  rootActor.tell(root -> ...)
  ...
  elg.await();
  rootActor.tell(root -> root.shutdownEverything) // will never be called.
}
```

If so then we can assume that the user would never do that in the first place. I think this is a safe assumption.

Do you see the benefit of adding the `await` method?

Reply:
Are you saying we can `await` on the main thread, then there's another thread issuing more events & stop the event loop?

Reply:
> Are you saying we can await on the main thread, then there's another thread issuing more events & stop the event loop?

Correct. Without doing an `await` on the main thread, the program will stop immediately - a computer program exits immediately when the main thread exits regardless of whether there are additional threads actively running in the background.